src/org/zaproxy/zap/extension/api/API.java

```diff
@@ -171,6 +180,11 @@ public boolean handleApiRequest (HttpRequestHeader requestHeader, HttpInputStrea
 		
 		// Check for callbacks
 		if (url.contains(CALL_BACK_URL)) {
+			if (! isPermittedIpAddr(requestHeader)) {
+				httpOut.close();
```

Comment:
This is not necessary, the connection is closed by the `ProxyThread` (same for other cases where it returns true). Or, is it to ensure that's closed?